modules/gapi/src/compiler/gcompiled.cpp

```diff
@@ -154,3 +156,16 @@ void cv::GCompiled::reshape(const GMetaArgs& inMetas, const GCompileArgs& args)
 {
     m_priv->reshape(inMetas, args);
 }
+
+void cv::GCompiled::setMetaData(ade::Graph &g, cv::GCompileArgs &args
+                                        , const cv::GMetaArgs &metas)
+{
+    auto pass_ctx = ade::passes::PassContext{g};
+    cv::gimpl::passes::initMeta(pass_ctx, metas);
+
+    cv::gimpl::passes::inferMeta(pass_ctx, true);
+    //compile islands for m_orig_graph
+    cv::gimpl::passes::storeResultingMeta(pass_ctx);
+    // Get compileArgs from m_ops??
+    cv::gimpl::GCompiler::compileIslands(g, args);
+}
```

Comment:
This file is a wrong place for this. `GCompiler`, not `GCompiled`, is the place for all compiler-related logic.